servicetalk-grpc-api/src/main/java/io/servicetalk/grpc/api/GrpcRouter.java

```diff
@@ -214,19 +256,37 @@ RouteProviders drainRoutes() {
                 final StreamingRoute<Req, Resp> route, final Class<Req> requestClass,
                 final Class<Resp> responseClass, final GrpcSerializationProvider serializationProvider) {
             streamingRoutes.put(path, new RouteProvider(executionContext -> {
-                StreamingHttpService service = (ctx, request, responseFactory) -> {
-                    try {
-                        final GrpcServiceContext serviceContext = new DefaultGrpcServiceContext(request.path(), ctx);
-                        final HttpDeserializer<Req> deserializer =
-                                serializationProvider.deserializerFor(readGrpcMessageEncoding(request), requestClass);
-                        final Publisher<Resp> response = route.handle(serviceContext, request.payloadBody(deserializer))
-                                .map(GrpcUtils::uncheckedCast);
-                        return succeeded(newResponse(responseFactory, response,
-                                serializationProvider.serializerFor(serviceContext, responseClass),
-                                ctx.executionContext().bufferAllocator()));
-                    } catch (Throwable t) {
-                        return succeeded(newErrorResponse(responseFactory, t,
-                                ctx.executionContext().bufferAllocator()));
+                StreamingHttpService service = new StreamingHttpService() {
+                    @Override
+                    public Single<StreamingHttpResponse> handle(final HttpServiceContext ctx,
+                                                                final StreamingHttpRequest request,
+                                                                final StreamingHttpResponseFactory responseFactory) {
+                        try {
+                            final GrpcServiceContext serviceContext =
+                                    new DefaultGrpcServiceContext(request.path(), ctx);
```

Comment:
future consideration for perf (/cc @jayv) ... `path()` takes some extra parsing and by this point we should have already gone through the router layer (which currently also does `path()`) but I wonder if we could just use the raw request target and avoid the parsing all together (and just bail if there are any additional components of the URL).

Reply:
We do cache the parsed result for the `URI` inside the request implementation[1] so calling `path()` multiple times would not incur additional parsing. Lets see if this shows up in perf and we can fix, thanks for pointing out 🙏 

[1] https://github.com/apple/servicetalk/blob/master/servicetalk-http-api/src/main/java/io/servicetalk/http/api/DefaultHttpRequestMetaData.java#L293-L295